contentful/apps
# Integ 310 listen for changes in status of published content to update render of warning message

## Purpose
This PR allows the GA4 sidebar app to listen for changes in the entry state of 'published' vs 'unpublished. This is important because we show a warning message in lieu of the app when the content is not published. If the user publishes the content, we then want to render the app appropriately. 

## Approach
The `sdk.entry` has a function called `onSysChanged` which is called when the entry sys metadata is adjusted in anyway, which includes the `publishedAt` attribute. Additionally, I made a few adjustments to assure that rendering the warning message and then subsequently rendering the app itself 'maintains' or 'clears' the appropriate state at the level of the `AnalyticsApp` component, such as the selected date range. The loading state was also adjusted to ensure the chart and any warning message render appropriately, and not before the actual gathering of data. Loading is fairly seamless now (from my local testing at least)🤞. Fixed some inconsistent margins and slight blurriness in the chart given the strict height requirement. 

## Results
https://www.loom.com/share/86238383a6f94426885156671530a4d9

## How to test
Render the GA4 app in the sidebar, and go ahead and toggle between publishing and un-publishing the entry content. This should toggle between rendering the warning message and the app itself. 

## Dependencies and/or References
[SDK docs](https://www.contentful.com/developers/docs/extensibility/app-framework/sdk/#entry)
